W-22107509 pdk upgrade gr (#1047)
mulesoft/docs-gateway: pdk/1.8/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc

----
. Save the `mod.rs` file.

[[upgrade-anypoint-cli-pdk-plugin-earlier-1.8.0]]
== Upgrade the Anypoint CLI PDK Plugin for Policies with Versions Earlier Than 1.8.0

> Depending on you configuration, when upgrading to 1.8.0 you might experience these issue:

Reviewer: ```suggestion
Depending on you configuration, when upgrading to 1.8.0 you might experience these issues:
```

* <<feature-compatibility-issue-on-tokio>>
* <<compilation-error-structs>>

[[feature-compatibility-issue-on-tokio]]
=== Tokio Runtime Feature Compatibility Error

> When using a Tokio runtime for testing, you might receive a compilation error similar to the following when building for WebAssembly:

Reviewer: ```suggestion
When using a Tokio runtime for testing, sometimes a compilation occurs similar to this error when building for WebAssembly:
```

[source,text]
----
error: Only features sync,macros,io-util,rt,time are supported on wasm.
   --> /Users/test-user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.51.1/src/lib.rs:478:1
    |
478 | compile_error!("Only features sync,macros,io-util,rt,time are supported on wasm.");
----

To solve this error, under the `[package]` section of your policy's `Cargo.toml`, add `resolver = "2"`. This enables your compiler resolve dependencies separately for production code and for test dependencies.

[source,toml]
----
[package]
name = "my_policy"
version = "1.0.0"
rust-version = "1.88.0"
edition = "2018"
resolver = "2"
----

[[compilation-error-structs]]
=== Compilation Error When Constructing Metadata-Related Structs

> When contruct Metadata-related structs, you might see an error similar to the following:

Reviewer: ```suggestion
When constructing Metadata-related structs, sometimes an error similar to this one occurs:
```

[source,text]
----
error[E0639]: cannot create non-exhaustive struct using struct expression
  --> test-policy/src/lib.rs:90:32
   |
90 |                   flex_metadata: FlexMetadata {
   |  ________________________________^
91 | |                     flex_name: "test_flex_name".to_string(),
92 | |                     flex_version: "1.0.0".to_string(),
93 | |                 },
   | |_________________^
----

> PDK 1.8.0 marks metadata-related structs as `#[non_exhaustive]` so you can add new metadata fields without breaking changes.

Reviewer: ```suggestion
PDK 1.8.0 marks metadata-related structs as `#[non_exhaustive]` so that you can add new metadata fields without breaking changes.
```
